control_protocol.md

```diff
@@ -0,0 +1,232 @@
+
+
+#### Definitions
+
+Zmq messages consist in a series of frames, each is a byte sequence.
+We indicate the separation between frames with `|`.
+An empty frame is indicated with two frame separators `||`, even at the beginning or end of a message.
+For example `||Second frame|Third frame||Fifth frame` consists of 5 frames.
+The first and fourth frames are empty frames.
+
+
+#### Configuration
+
+Each {ref}`Coordinator <components.md#coordinator>` shall offer one ROUTER socket, bound to a host name (or IP address or any address of a computer with "*") and port in the DMT.
+
+{ref}`Components <components.md#components>` shall have one DEALER socket connecting to one Coordinator's ROUTER socket.
+
+Coordinators shall have one DEALER socket per other Coordinator in the network.
+This DEALER socket shall connect to the other Coordinator's ROUTER socket.
+
+Messages must be sent to a Coordinator's ROUTER socket.
+
+
+#### Particularities of ROUTER sockets
+
+A small introduction to ROUTER sockets, for more details see [zmq guide chapter 3](https://zguide.zeromq.org/docs/chapter3/#Exploring-ROUTER-Sockets).
+
+A router socket assigns a random identity to each connecting peer.
+If, for example, two Components `CA`, `CB` connect to the ROUTER sockets, the socket assigns the identities, which we call `IA`, `IB` (they can be any bytes sequence).
+
+Whenever a message is sent to the ROUTER socket from a peer, it reads as a first frame that identity.
+For example if `CA` sends the message `Request A`, the ROUTER socket will read `IA|Request A`.
+That way, you can return an answer to exactly the original peer and not, for example `CB`.
+If you call the ROUTER's send command with `IA|Reply A`, the socket will send `Reply A` to the peer, whose connection is `IA`, in this case `CA`.
+
+The following diagram shows this example communication with two Components:
+:::{mermaid}
+graph TD
+    CA([CA]) -->|"Request A"| D[recv] -->|"IA|Request A"| A([Coordinator])
+    CB([CB]) -->|"Request B"| C[recv] -->|"IB|Request B"| A
+    subgraph ROUTER socket
+        B
+        C
+        D
+        E
+    end
+    A -. "IA|Reply A" .-> B
+    B[send] -. "Reply A" .-> CA
+    A -. "IB|Reply B" .-> E
+    E[send] -. "Reply B" .-> CB
+:::
+
+
+### Naming scheme
+
+Each Component has an individual ID, given by the user, the Component ID.
+A Component ID must be a series of bytes, without the ASCII character "." (byte value 46).
+Component IDs must be unique in a Node, i.e. among the Componets connected to a single Coordinator.
+
+As each Component belongs to exactly one Coordinator, it is fully identified by the combination of Node ID and Component ID.
+This _full ID_ is the composition of Node ID, ".", and Component ID.
+
+The receiver of a message may be specified without the Node ID, if the receiver lives in the same Node.
+
+:::{note}
+How to address the Coordinator? I used "no Recipient".
+Or a fixed name?
+:::
```

Comment:
I am not sure I understand, this is now about what to do when I want to talk to the Coordinator itself? maybe repeat the Node ID? CA wants to talk directly to Co1 (for whatever reason): `| Co1.Co1 | Co1.CA | other frames  ` - when Co1 tries to look up Co1 it finds itself in its own address book

Reply:
I added `COORDINATOR` as a placeholder address, such that it is easy to replace all occurances at once.